modules/tracing/mule-tracer-exporter-impl/src/main/java/org/mule/runtime/tracer/impl/exporter/OpenTelemetrySpanExporter.java

```diff
@@ -104,42 +232,186 @@ public void updateChildSpanExporter(SpanExporter childSpanExporter) {
+      }
+
+      if (childOpenTelemetrySpanExporter.parentSpanContext == getInvalid()) {
+        childOpenTelemetrySpanExporter.parentSpanContext = spanContext;
       }
     }
+  }
 
+  @Override
+  public InternalSpan getInternalSpan() {
+    return internalSpan;
   }
 
-  private void setRootSpan(OpenTelemetrySpanExporter openTelemetrySpan) {
-    this.rootSpan = openTelemetrySpan;
+  @Override
+  public String getName() {
+    if (overriddenSpanName == null) {
+      return internalSpan.getName();
+    }
+
+    return overriddenSpanName;
   }
 
-  private String getWithoutNamespace(String name) {
-    int index = name.lastIndexOf(":");
-    if (index != -1) {
-      return name.substring(index + 1);
-    } else {
-      return name;
+  @Override
+  public SpanKind getKind() {
+    return spanKind;
+  }
+
+  @Override
+  public void onAdditionalAttribute(String key, String value) {
+    if (key.equals(SPAN_KIND)) {
+      spanKind = SpanKind.valueOf(value);
+    }
+
+    if (key.equals(STATUS)) {
+      StatusCode statusCode = StatusCode.valueOf(value);
+      statusData = StatusData.create(statusCode, null);
     }
   }
 
   @Override
-  public InternalSpan getInternalSpan() {
-    return internalSpan;
+  public SpanContext getSpanContext() {
+    return spanContext;
+  }
+
+  @Override
+  public SpanContext getParentSpanContext() {
+    return parentSpanContext;
+  }
+
+  @Override
+  public String getParentSpanId() {
+    return SpanData.super.getParentSpanId();
   }
 
+  @Override
+  public StatusData getStatus() {
+    return statusData;
+  }
+
+  @Override
+  public long getStartEpochNanos() {
+    return internalSpan.getDuration().getStart();
+  }
+
+  @Override
+  public Attributes getAttributes() {
+    return this;
+  }
+
+  @Override
+  public List<EventData> getEvents() {
+    return errorEvents;
+  }
+
+  @Override
+  public void onError(InternalSpanError spanError) {
+    statusData = StatusData.create(ERROR, EXCEPTIONS_HAS_BEEN_RECORDED);
+    Attributes errorAttributes = of(EXCEPTION_TYPE_KEY, spanError.getError().getErrorType().toString(),
+                                    EXCEPTION_MESSAGE_KEY, spanError.getError().getDescription(),
+                                    EXCEPTION_STACK_TRACE_KEY,
+                                    getInternalSpanError(spanError).getErrorStacktrace().toString(),
+                                    EXCEPTION_ESCAPED_KEY, spanError.isEscapingSpan());
+
+    errorEvents = singletonList(EventData.create(System.currentTimeMillis(), EXCEPTION_EVENT_NAME, errorAttributes));
+  }
+
+  @Override
+  public List<LinkData> getLinks() {
+    return emptyList();
+  }
+
+  @Override
+  public long getEndEpochNanos() {
+    return internalSpan.getDuration().getStart();
+  }
+
+  @Override
+  public boolean hasEnded() {
+    return true;
+  }
+
+  @Override
+  public int getTotalRecordedEvents() {
+    if (errorEvents.isEmpty()) {
+      return 0;
+    }
+
+    return 1;
```

Comment:
returning 1 instead of errorEvents.size() is due to performance?

Reply:
yes, because we know that in the current implementation we have only 1 error per span. I'll add a comment and a constant.